esbuild.js

```diff
@@ -0,0 +1,127 @@
+
+const src = './src';
+const argsObj = yargs(argv).argv;
+
+const electronEntryFiles = {
+    renderer: [path.resolve(__dirname, 'src/electron/views/renderer-initializer.ts')],
+    main: [path.resolve(__dirname, 'src/electron/main/main.ts')],
+};
+const unifiedOutdir = path.join(__dirname, 'extension/unifiedBundle');
+
+const webExtensionEntryFiles = {
+    background: `${src}/background/background-init.ts`,
+    injected: `${src}/injected/client-init.ts`,
+    popup: `${src}/popup/popup-init.ts`,
+    insights: `${src}/views/insights/initializer.ts`,
+    detailsView: `${src}/DetailsView/details-view-initializer.ts`,
+    devtools: `${src}/Devtools/dev-tool-init.ts`,
+    serviceWorker: `${src}/background/service-worker-init.ts`,
+    debugTools: `${src}/debug-tools/initializer/debug-tools-init.tsx`,
+};
+const devWebExtensionOutdir = path.join(__dirname, 'extension/devBundle');
+const devWebExtensionM3Outdir = path.join(__dirname, 'extension/devMv3Bundle');
+
+const prodWebExtensionOutDir = path.join(__dirname, 'extension/prodBundle');
+
+// Default behavior; builds dev extension.
+let entryFiles = webExtensionEntryFiles;
+let outdir = devWebExtensionOutdir;
+let isProd = false;
+let platform = 'browser';
+let external = [];
+let format = 'esm';
+let define;
+let target = ['chrome90', 'firefox90'];
+let plugins = [CreateStylePlugin()];
+
+switch (argsObj.env) {
+    // Note: currently causes errors when electron app is run.
+    case 'unified':
+        entryFiles = electronEntryFiles;
+        outdir = unifiedOutdir;
+        platform = 'node';
+        external = [];
+        format = 'cjs';
+        break;
+
+    case 'prod':
+        isProd = true;
+        outdir = prodWebExtensionOutDir;
+        delete entryFiles.detailsView;
+        break;
+
+    case 'devM3':
+        outdir = devWebExtensionM3Outdir;
+        define = {
+            global: 'globalThis',
+        };
+        break;
+
+    case 'report':
+        entryFiles = { report: `${src}/reports/package/reporter-factory.ts` };
+        outdir = path.join(__dirname, 'packages/report/bundle');
+
+        // esbuild doesn't have an easy way to ignore node_modules for monorepos,
+        // so this plugin is necessary (marks all node_modules as external).
+        // Thread: https://github.com/evanw/esbuild/issues/619
+        plugins.push(
+            NodeResolve.NodeResolvePlugin({
+                extensions: ['.ts', '.tsx', '.js'],
+                onResolved: resolved => {
+                    if (resolved.includes('node_modules')) {
+                        return {
+                            external: true,
+                        };
+                    }
+                    return resolved;
+                },
+            }),
+        );
+        break;
+}
+
+const config = {
+    plugins,
+    entryPoints: entryFiles,
+    format,
+    platform,
+    external,
+    outbase: src,
+    bundle: true,
+    minify: isProd,
```

Comment:
For other reviewers: The MPL requires us to preserve axe-core's license notice even in our minified prod bundles. When I was reviewing, I saw that esbuild's prod bundles didn't have the same legal comments at the top of the file that webpack's prod bundles did and thought this would be a problem, but it turns out that [esbuild just defaults to putting them at the end of the file](https://esbuild.github.io/api/#legal-comments) instead of the beginning (this is fine).